pkg/zkcertificate/certificate_simple_json.go

```diff
@@ -0,0 +1,132 @@
+package zkcertificate
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"sort"
+
+	"github.com/iden3/go-iden3-crypto/poseidon"
+)
+
+// SimpleJSON represents the input data for data that consists of
+// simple JSON fields: strings, numbers, and booleans.
+type SimpleJSON map[string]interface{}
+
+// FFEncode implements FFEncoder.
+func (k SimpleJSON) FFEncode() (SimpleJSONContent, error) {
+	keys := make([]string, 0, len(k))
+	for key := range k {
+		keys = append(keys, key)
+	}
+	sort.Strings(keys)
+
+	hashedContent := make(SimpleJSONContent, len(k))
+
+	for i, key := range keys {
+		value := k[key]
+		var valueStr string
+
+		switch v := value.(type) {
+		case string:
+			valueStr = v
+		case int, int8, int16, int32, int64, uint, uint8, uint16, uint32, uint64, float32, float64:
+			valueStr = fmt.Sprintf("%v", v)
+		case bool:
+			valueStr = fmt.Sprintf("%t", v)
+		default:
+			return nil, fmt.Errorf("unsupported type for field %s: %T", key, v)
+		}
+
+		hash, err := poseidon.HashBytes([]byte(valueStr))
+		if err != nil {
+			return nil, fmt.Errorf("hash field %s: %w", key, err)
+		}
+		hashedContent[i] = HashFromBigInt(hash)
+	}
+
+	return hashedContent, nil
+}
+
+// Validate performs validation on the SimpleJSON instance.
+func (c *SimpleJSON) Validate() error {
+	if c == nil {
+		return fmt.Errorf("cannot validate nil SimpleJSON")
+	}
+
+	keys := make([]string, 0, len(*c))
+	for key := range *c {
+		keys = append(keys, key)
+	}
+	sort.Strings(keys)
+
+	for _, key := range keys {
+		value := (*c)[key]
+		var valueStr string
+
+		switch v := value.(type) {
+		case string:
+			valueStr = v
+		case int, int8, int16, int32, int64, uint, uint8, uint16, uint32, uint64, float32, float64:
+			valueStr = fmt.Sprintf("%v", v)
+		case bool:
+			valueStr = fmt.Sprintf("%t", v)
+		default:
+			return fmt.Errorf("unsupported type for field %s: %T", key, v)
+		}
+
+		_, err := poseidon.HashBytes([]byte(valueStr))
```

Comment:
`Validate` should validate that your type is valid. If it can't be hashed, it's a different problem. I suggest not calling this here. And hence don't bother sorting keys here either